tests/test_impedance.py

```diff
@@ -0,0 +1,235 @@
+    with pytest.raises(ValueError):
+        pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[2, 2, -3.2], channel_phase=[-np.inf, 2, 3.3])
+    with pytest.raises(ValueError):
+        pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[2, 2, -3.2], channel_phase=[np.inf, 2, 3.3])
+
+    # check None inputs
+    impedance_object = pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[1, -2.1, -3.2], channel_phase=[1, 2, 3.3])
+    assert impedance_object.channel_label is None
+    assert impedance_object.channel_unit is None
+    assert impedance_object.channel_color is None
+    assert impedance_object.channel_source is None
+    assert impedance_object.channel_linestyle is None
+
+    # check inputs
+    impedance_object = pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[1, -2.1, -3.2], channel_phase=[1, 2, 3.3],
+                                                               channel_label="test 1", channel_unit="A", channel_color="red",
+                                                               channel_source="scope 11", channel_linestyle="--")
+    assert impedance_object.channel_label == "test 1"
+    assert impedance_object.channel_unit == "A"
+    assert impedance_object.channel_color == "red"
+    assert impedance_object.channel_source == "scope 11"
+    assert impedance_object.channel_linestyle == "--"
+
+    # wrong type inputs
+    with pytest.raises(TypeError):
+        pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[1, -2.1, -3.2],
+                                                channel_phase=[1, 2, 3.3], channel_label=100.1)
+    with pytest.raises(TypeError):
+        pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[1, -2.1, -3.2],
+                                                channel_phase=[1, 2, 3.3], channel_unit=100.1)
+    with pytest.raises(TypeError):
+        pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[1, -2.1, -3.2],
+                                                channel_phase=[1, 2, 3.3], channel_color=100.1)
+    with pytest.raises(TypeError):
+        pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[1, -2.1, -3.2],
+                                                channel_phase=[1, 2, 3.3], channel_source=100.1)
+    with pytest.raises(TypeError):
+        pss.Impedance.generate_impedance_object(channel_frequency=[1, 2, 3], channel_impedance=[1, -2.1, -3.2],
+                                                channel_phase=[1, 2, 3.3], channel_linestyle=100.1)
+
+
+def test_save_load():
+    """Unit test for save and load."""
+    # assumption: the given scope object is valid
+    example = pss.Impedance.generate_impedance_object([1, 2, 3], [4, 5, 6],
```

Comment:
Better is to use a scope object with mixed data to check, if the different types of data and numbers (Positiv  and negative integer and float values) are stored and reload correctly.
E.g. [0.1,1,1.2,2.1,3], [0,5.1,1.2,2.1,3], phase: [-4,-5.9,0,6,1.2] 
Assumption: Frequency and impedance values shall be only positive, phase values are float or int?

Reply:
done